### PR TITLE
Added kafka producer code as a singleton.

### PR DESCRIPTION
**Pull request by AaronH88**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by loadtheaccumulator — COMMENTED** [review]
Should this be in a common pkg to be easily used for routes, services, and the microservices?

**Comment by AaronH88** [comment]
This needs to be merged after #1224 as the mods have been fixed in 1224

**Comment by AaronH88** [comment]
> Should this be in a common pkg to be easily (or more obviously) used for routes, services, and the microservices?

Any suggestions?

**Comment by AaronH88** [comment]
will move to `pkg/common/kafka`

File: pkg/common/kafka/producer.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+func getInstance() *kafka.Producer {
+	if singleInstance == nil {
+		lock.Lock()
+		defer lock.Unlock()
+		if singleInstance == nil && clowder.IsClowderEnabled() {
+			brokers := make([]string, len(clowder.LoadedConfig.Kafka.Brokers))
+			for i, b := range clowder.LoadedConfig.Kafka.Brokers {
+				brokers[i] = fmt.Sprintf("%s:%d", b.Hostname, *b.Port)
+			}
+			p, err := kafka.NewProducer(&kafka.ConfigMap{
```
Comment (loadtheaccumulator) [inline pkg/common/kafka/producer.go:25]: @AaronH88 We'll need to include auth info here as well.
Similar to this... https://github.com/grandcentralstation/kafkavents-junit-client/blob/755947996403dc88c65ddbaf61ec4640e04799f3/src/kafkavents/kafka.go#L81-L95

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]
Some bits need to be added for auth.

**Review by loadtheaccumulator — APPROVED** [review]
lgtm

**Pull request merged**